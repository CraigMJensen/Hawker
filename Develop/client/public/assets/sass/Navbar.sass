@import './_variables'
@import './_responsive'
@import './index'

nav
  @include flex
  width: 100%

.links
  @include flex-right
  list-style: none
  text-decoration: none

.links li a
  margin: 1vw

#logo
  @include flex-left
  padding-top: 0
  margin-top: 0
