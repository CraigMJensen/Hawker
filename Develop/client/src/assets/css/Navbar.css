/* VARIABLES */
/* Foreground Colors */
/* Background Colors */
/* Spacing */
/* GENERAL STYLES */
/* VARIABLES */
/* Foreground Colors */
/* Background Colors */
/* Spacing */
* {
  -webkit-box-sizing: border-box;
  box-sizing: border-box;
}

body {
  margin: 0;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen", "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  margin: 0;
  padding: 0;
  font-weight: 600;
  line-height: 1.5;
  font-size: 18px;
  letter-spacing: 0.1em;
  color: #023047;
}

html,
body {
  min-height: 100%;
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New", monospace;
}

main {
  margin: 0 auto;
  max-width: 90%;
}

nav {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
}

.links {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: flex-start;
  list-style: none;
  text-decoration: none;
}

.links li a {
  margin: 1vw;
}

#logo {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: flex-start;
  padding-top: 0;
  margin-top: 0;
}

/*# sourceMappingURL=Navbar.css.map */

/*# sourceMappingURL=Navbar.css.map */
