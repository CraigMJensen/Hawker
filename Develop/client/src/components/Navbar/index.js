//TODO: NAVIGATION COMPONENT
//! Import dependencies
import React from 'react';
import { Link } from 'react-router-dom';
import Auth from '../../utils/auth';
import Hawker from '../../assets/images/icons/Hawker.svg';
import '../../assets/css/Navbar.css';

//! Create Navigation component
function Nav() {
  function showNavigation() {
    if (Auth.loggedIn()) {
      return (
        <>
          <li>
            <Link to="/orderHistory">Order History</Link>
          </li>
          <li>
            {/* this is not using the Link component to logout or user and then refresh the application to the start */}
            <a href="/" onClick={() => Auth.logout()}>
              Logout
            </a>
          </li>
        </>
      );
    } else {
      return (
        <>
          <li>
            <Link to="/signup">Signup</Link>
          </li>
          <li>
            <Link to="/login">Login</Link>
          </li>
        </>
      );
    }
  }

  return (
    <nav>
      <span>
        <Link to="/">
          <span role="img" aria-label="img-name">
            <img src={Hawker} alt="logo" className="img-thumbnail mx-1 inline-flex" id="logo" />
          </span>
        </Link>
      </span>

      <ul className="links">{showNavigation()}</ul>
    </nav>
  );
}

export default Nav;
