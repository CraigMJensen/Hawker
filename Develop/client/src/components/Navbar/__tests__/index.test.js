// Import dependencies
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Navbar from '..';

// After each test, make sure there isnt any leftover memory that could give you false results
afterEach(cleanup);

// Declare the components you are testing
describe('Navbar component', () => {
  // verify the component is rendering properly
  it('Navbar component renders', () => {
    render(<Navbar />);
  });

  // compare snapshot versions of the DOM node structure
  it('matches snapshot DOM node structure', () => {
    const { asFragment } = render(<Navbar />);
    expect(asFragment()).toMatchSnapshot();
  });
});