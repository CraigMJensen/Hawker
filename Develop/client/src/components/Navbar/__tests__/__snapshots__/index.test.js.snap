// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <div>
    <h2>
      Navbar
    </h2>
  </div>
</DocumentFragment>
`;
